src/v/kafka/client/client.cc

```diff
@@ -172,6 +172,21 @@ ss::future<> client::mitigate_error(std::exception_ptr ex) {
               case error_code::coordinator_not_available:
                   vlog(kclog.debug, "consumer_error: {}", ex);
                   return _wait_or_start_update_metadata();
+              case error_code::not_coordinator:
+                  vlog(
+                    kclog.info,
+                    "consumer_error: {}, running find coordinator before retry",
+                    ex);
+                  return do_find_coordinator(ex.group_id, ex.member_id)
+                    .then([this, group_id{ex.group_id}, name{ex.member_id}](
+                            shared_broker_t new_coordinator) mutable {
+                        return get_consumer(group_id, name)
+                          .then([new_coordinator{new_coordinator}](
+                                  shared_consumer_t c) {
+                              c->set_coordinator(new_coordinator);
+                              return ss::now();
+                          });
+                    });
```

Comment:
After I pushed I realized some weaknesses with this approach.
I think a problem with this first approach is that:
1. If `not_coordinator` is thrown on the last retry (perhaps we were already retrying for a different error), then the original request will
fail because another retry did not occur to run on the new coordinator
2. This change only affects those kafka requests that run on the coordinator; however, within the consumer class itself, there is administrative code (`consumer::join`, `consumer::sync`, `consumer::heartbeat`) that uses the coordinator via `consumer::req_res`. So those places also need to account for
not_coordinator error. 

A different solution is to:
1. put gated_retry_with_mitigation in a separate location so it can be called from anywhere
2. Create a do_find_coordinator method in the consumer
3. Mod `consumer::red_res` to to handle `not_coordinator` errors from responses

but the problem with (3) is that a kafka error_code can be in different response levels depending on the Kafka request.
For example, `offset_fetch_response` puts the error code in the top level of the response but `offset_commit_response` puts errors at the partition level.

Reply:
Perhaps I can do a type check for the `response_t`. The default can assume errors are at the top level and then we can look through the proper levels for different response types such as `offset_commit_response`